internal/store/postgres/resource/repository.go

```diff
@@ -53,27 +53,81 @@ func (r Repository) Update(ctx context.Context, resourceModel *resource.Resource
 	return nil
 }
 
+func (r Repository) Delete(ctx context.Context, resourceModel *resource.Resource) error {
+	res := FromResourceToModel(resourceModel)
+
+	deleteResourceQuery := `UPDATE resource SET status=$5, updated_at=NOW()
+                WHERE full_name=$1 AND store=$2 AND project_name = $3 And namespace_name = $4`
+	tag, err := r.db.Exec(ctx, deleteResourceQuery, res.FullName, res.Store, res.ProjectName, res.NamespaceName, resource.StatusDeleted)
+	if err != nil {
+		return errors.Wrap(resource.EntityResource, "error updating resource to database", err)
+	}
+
+	if tag.RowsAffected() == 0 {
+		return errors.NotFound(resource.EntityResource, "no resource to delete for "+res.FullName)
+	}
+	return nil
+}
+
 func (r Repository) ChangeNamespace(ctx context.Context, res *resource.Resource, newTenant tenant.Tenant) error {
+	existingResource, err := r.ReadByFullName(ctx, newTenant, res.Store(), res.FullName(), false)
+	if err != nil && !errors.IsErrorType(err, errors.ErrNotFound) {
+		return err
+	}
+
+	tx, err := r.db.Begin(ctx)
+	if err != nil {
+		return errors.Wrap(resource.EntityResource, "error begin db transaction", err)
+	}
+
+	err = r.hardDelete(ctx, tx, existingResource)
+	if err != nil {
+		_ = tx.Rollback(ctx)
+		return err
+	}
+
 	updateResource := `UPDATE resource SET namespace_name=$1, updated_at=now()
 	WHERE full_name=$2 AND store=$3 AND project_name = $4 And namespace_name = $5`
-	tag, err := r.db.Exec(ctx, updateResource,
+	tag, err := tx.Exec(ctx, updateResource,
 		newTenant.NamespaceName(), res.FullName(), res.Store(),
 		res.Tenant().ProjectName(), res.Tenant().NamespaceName())
 	if err != nil {
+		_ = tx.Rollback(ctx)
 		return errors.Wrap(resource.EntityResource, "error changing tenant for resource:"+res.FullName(), err)
 	}
-
 	if tag.RowsAffected() == 0 {
+		_ = tx.Rollback(ctx)
 		return errors.NotFound(resource.EntityResource, "no resource to changing tenant for ")
 	}
-	return nil
+
+	err = tx.Commit(ctx)
+	return errors.WrapIfErr(resource.EntityResource, "error commit db transaction", err)
+}
+
+func (Repository) hardDelete(ctx context.Context, tx pgx.Tx, res *resource.Resource) error {
+	if res == nil || !res.IsDeleted() {
```

Comment:
have we properly handled if the resource exists in the destination namespace but is still active?

Reply:
we will not do hard delete, if its still exists. Any particular concern about this ?

Reply:
okay, we can keep the same behavior. it should not be hard deleted.